coralproject/talk
# [CORL-2355]: Z key go to next unread comment/reply on view more/new comments and replies

## What does this PR do?

These changes update the Z key behavior so that when it hits "Load More", "Show All Replies", "View New Comment", and "Show More Replies", it focuses on the next unread comment, instead of stopping on the (previously read) comment next to the button (which might be far from the next unread). These changes are for https://vmproduct.atlassian.net/browse/CORL-2355.

## What changes to the GraphQL/Database Schema does this PR introduce?

None.

## How do I test this PR?

You can test that new Z key functionality works as expected. As User A who hasn't read anything in the stream, use Z key to navigate through. Log in as User B and add comments above and below `Load More` pagination. As User A, return to the stream and see that Z key correctly navigates through the new unread comments, even when they are below and far from the `Load More` button. As User A, add new unread replies to a comment, enough to have the `Show All Replies` button show up. As User B, navigate through using the Z key and see that it correctly navigates to the new unread replies even when the button is hit. As User A, while you have User B open in another tab, add new comments and new replies to comments. See the "View New Comments" and "Show More Replies" buttons show up for User B. As User B, navigate with the Z key and see that it correctly navigates through these buttons to show the new unread comments/replies.

You can also test that all other shortcut functionality still works as expected. So, C key should still scroll you forwards through the comments and shift+C backwards through the comments. shift+A should still mark all comments on screen as read.
 
## How do we deploy this PR?

None.
